### PR TITLE
Fixed : Navbar Make it Work

### PR DESCRIPTION
**Pull request by Rishi05051997**
Peer Members, Navbar issue is fixed

### CODE REVIEW

**Review by JowelTisso — APPROVED** [review]
* Clean code looks good

File: src/components/Layout/Nav.jsx [hunk 1]
```diff
@@ -1,8 +1,24 @@
 import { Icon } from "@iconify/react";
 import header from "./../../../src/assets/apna-5.PNG";
 import { Link } from "react-router-dom";
+import { useData } from "../../context/data-context";
+import { useState } from "react";
+import { useAuthContext } from "../../context";
 
 const Nav = () => {
+    const { login, logOutUser } = useAuthContext()
+    console.log(login)
```
Comment (ivinayakg) [inline src/components/Layout/Nav.jsx]: remove console.log

File: src/components/Layout/Nav.jsx [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
+    const { firstName } = login;
+    const [searchTxt, setSearchTxt] = useState("");
+    const {
+        state: { itemsInCart, itemsInWishlist }, dispatch } = useData();
+    const searchHandler = (e) => {
+        if (e.keyCode === 13) {
```
Comment (ivinayakg) [inline src/components/Layout/Nav.jsx:15]: whats this?

File: src/components/Layout/Nav.jsx [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+            dispatch({ type: "SEARCH_PRODUCT", payload: searchTxt });
```
Comment (ivinayakg) [inline src/components/Layout/Nav.jsx:16]: try doing this search functionality from a useParams

File: src/components/Layout/Nav.jsx [hunk 4]
```diff
@@ -34,38 +50,49 @@ const Nav = () => {
...
-                        <span className="iconify icons text-2">
+                        <span onClick={() => {
+                            dispatch({ type: "SEARCH_PRODUCT", payload: searchTxt });
+                            setSearchTxt("");
+                        }} className="iconify icons text-2">
                             <Icon icon="ant-design:search-outlined" />
                         </span>
 
-                        <label htmlFor="name" className="input-label text-2">
+                        <label for="name" className="input-label text-2">
```
Comment (rahulyadav139) [inline src/components/Layout/Nav.jsx]: in react's JSX htmlFor is used for "for" attribute. Other than that everything looks great. Go ahead and merge the branch.

**Review by rahulyadav139 — APPROVED** [review]

**Review by ivinayakg — APPROVED** [review]
made a suggestion for the implementation for the functionality of search bar, try to do it,

File: src/components/Layout/Nav.jsx (hunk 2 above)
Comment (Rishi05051997) [reply src/components/Layout/Nav.jsx:15]: While searching I'm searching keycode

File: src/components/Layout/Nav.jsx (hunk 3 above)
Comment (Rishi05051997) [reply src/components/Layout/Nav.jsx:16]: I will try

**Pull request merged**